### ISSUE TITLE
Parser fail when yield-ing an import() statement

### ISSUE THREAD

**Issue by Holi0317** [webpack/webpack#3925]
**Do you want to request a *feature* or report a *bug*?**
Report a bug

**What is the current behavior?**
Webpack parser will fail when it parse `yeild import('module')`.

**If the current behavior is a bug, please provide the steps to reproduce.**
Using `node_modules/.bin/webpack --entry='./generator' --output-filename='out.js'` for generating output. lodash is already installed.

Consider the following code
```js
function* main() {
    yield import('lodash');
}
```
The following error will be produced:
```
ERROR in ./generator.js
Module parse failed: /tmp/webpack-test/generator.js Unexpected token (2:8)
You may need an appropriate loader to handle this file type.
| function* main() {
|   yield import('lodash');
| }
| 
```
But this will pass
```js
function* main() {
  let i = import('lodash');
  yield i;
}
```
This also pass
```js
async function main() {
  await import('lodash')
}
```

**What is the expected behavior?**
Directly yield-ing an import() statement should not throw an error.

**If this is a feature request, what is motivation or use case for changing the behavior?**

**Please mention other relevant information such as the browser version, Node.js version, Operating System and programming language.**
Webpack version: 2.2.0-rc.4
Node version: 7.3.0

**Comment by sokra** [comment]
Looks like a bug in the acorn plugins. @kesne

**Comment by kesne** [comment]
Yeah looks like an issue with this: https://github.com/kesne/acorn-dynamic-import

I'll take a look in the morning, and also make sure that babel works as expected with yield statements too.

**Comment by kesne** [comment]
This is also happening in the babel parser.

**Comment by alexeyraspopov** [comment]
I've just run into the same issue. Any updates here?

Also, there is a question, what should be returned from `import()` as a module instance?

I have a module which is defined like:

```javascript
import React from 'react';

export default function App() {
  return (
    <h1>Hello World</h1>
  );
}
```

and when I try to import it dynamically, I have to do next:

```javascript
async function AppLoader() {
  // Issue described on this page
  const module = import('./App');
  // by default I receive an object with default field.
  // Though I was expecting actual default export
  const { default: App } = await module;
  return <App />;
}
```

Any ideas what can be wrong here?

**Comment by alexeyraspopov** [comment]
Seems like I've found the answer https://github.com/webpack/webpack/issues/3829

**Comment by Kovensky** [comment]
@alexeyraspopov `import()` returns the `[[ModuleRecord]]` object; so it's similar to a `Promise` version of `import * as name from 'module'`.

If it was not like that, it would be impossible to access exports other than the `default` export of a module, and there are many valid modules without a `default` export that may have other named exports.

**Comment by alexeyraspopov** [comment]
I totally agree with this behavior. I figured the thing with `import * as name from 'module'`. Sorry for the offtopic here

**Comment by kesne** [comment]
This will be fixed here: https://github.com/kesne/acorn-dynamic-import/pull/5.

It was also fixed in Babel, so if you're using babel-eslint, it should lint correctly.

**Comment by kesne** [comment]
This was landed as `acorn-dynamic-import@2.0.2`.

**Comment by alexeyraspopov** [comment]
I can confirm that the issue is fixed. Thanks @kesne
